docs/concepts/glossary.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Glossary
+nav-pos: 3
+nav-title: Glossary
+nav-parent_id: concepts
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+#### Flink Application Cluster
+
+A Flink Application Cluster is a dedicated [Flink Cluster](./glossary#flink-cluster) that only
+executes a single [Flink Job](./glossary#flink-job). The lifetime of the
+[Flink Cluster](./glossary#flink-cluster) is bound to the lifetime of the Flink Job. Formerly
+Flink Application Clusters were also known as Flink Clusters in *job mode*. Compare to
+[Flink Session Cluster](./glossary#flink-session-cluster).
+
+#### Flink Cluster
+
+The distributed system consisting of (typically) one Flink Master process and one or more Flink
+Taskmanagers processes.
+
+#### Event
+
+An event is a statement about a change of the state of the domain modelled by the
+application. Events can be input and/or output of a stream or batch processing application.
+Events are special types of [records](./glossary#Record).
```

Comment:
I find this rather complex: `a change of the state of the domain modelled by the
application` and honestly don't know what that means. Maybe rephrase but also give an example (not just "special type or record")

Reply:
In Domain Driven Design, Event-Driven Architectures, etc. this is exactly, what an event is. It is always an event (=state change) in the domain. I could add a text book example like "checkout" or "item added to cart". Would this help?